### PR TITLE
drivers: rtc: add ambiq rtc driver

### PR DESCRIPTION
**Pull request by RichardSWheatley**
Add Ambiq RTC driver

Signed-off-by: Richard Wheatley <richard.wheatley@ambiq.com>

### CODE REVIEW

File: tests/drivers/rtc/rtc_api/boards/apollo4p_evb.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2023 Bjarki Arge Andreasen
```
Comment (AlessandroLuo) [inline tests/drivers/rtc/rtc_api/boards/apollo4p_evb.conf:1]: wrong company name

File: drivers/rtc/Kconfig.ambiq [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2023 Linumiz
+# Author: Sri Surya  <srisurya@linumiz.com>
```
Comment (AlessandroLuo) [inline drivers/rtc/Kconfig.ambiq:3]: pls fix the infos

**Review by AlessandroLuo — CHANGES_REQUESTED** [review]

**Review by bjarki-andreasen — COMMENTED** [review]
Nice to see the driver finally appear :)

File: dts/arm/ambiq/ambiq_apollo4p.dtsi [hunk 3]
```diff
@@ -318,6 +318,13 @@
 			ambiq,pwrcfg = <&pwrcfg 0x4 0x10000>;
 		};
 
+		rtc0: rtc@40004800 {
+			compatible = "ambiq,rtc";
+			reg = <0x40004800 0x210>;
+			interrupts = <2 0>;
+			status = "disabled";
```
[the remark this answers is not in the document]
Comment (RichardSWheatley) [reply dts/arm/ambiq/ambiq_apollo4p.dtsi:325]: Fixed.

**Review by bjarki-andreasen — DISMISSED** [review]
Nice!

**Review by bjarki-andreasen — DISMISSED** [review]

**Comment by RichardSWheatley** [comment]
The rtc/shell twister test is failing.

412/744 apollo4p_evb              tests/drivers/rtc/shell/drivers.rtc.shell           ERROR Build failure (build)

This does not seem to be related to my driver but I cannot figure out where to go from here.

**Comment by bjarki-andreasen** [comment]
The CI failure is twofold, first of, the fake used for the shell is quite broken if CONFIG_ALARM or CONFIG_UPDATE is selected. This is fixed with this PR #76774

Second, you enabling CONFIG_ALARM if the ambiq driver is enabled. The alarm feature is optional, an its not intended to be included unless explicitly enabled by the application :) Please remove
```
config RTC_ALARM
	default y if RTC_AMBIQ
```
and it will work even without the fix in #76774 :)

**Review by aaronyegx — DISMISSED** [review]

**Comment by RichardSWheatley** [comment]
> config RTC_ALARM
> 	default y if RTC_AMBIQ

In this case, I left it but put _default n if RTC_AMBIQ_ to be explicit.

**Comment by bjarki-andreasen** [comment]
> > config RTC_ALARM
> > 	default y if RTC_AMBIQ
> 
> In this case, I left it but put _default n if RTC_AMBIQ_ to be explicit.

That was already the case, nothing enables the RTC_ALARM config by default, you can really just remove it :)

**Review by AlessandroLuo — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Pull request merged**